apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/FdaDrugCodeDisplayLookup.java

```diff
@@ -0,0 +1,179 @@
+
+import gov.cms.bfd.server.war.FDADrugDataUtilityApp;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Provides an FDA Drug Code to FDA Drug Code Display lookup */
+public class FdaDrugCodeDisplayLookup {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FdaDrugCodeDisplayLookup.class);
+
+  /** A fake drug code used for testing */
+  public static final String FAKE_DRUG_CODE = "00000-0000";
+
+  /** A fake drug code display that is associated with the FAKE_DRUG_CODE */
+  public static final String FAKE_DRUG_CODE_DISPLAY = "Fake Diluent - WATER";
+
+  /**
+   * Stores a map from Drug Code (PRODUCTNDC) to Drug Code Display (SUBSTANCENAME) derived from the
+   * downloaded NDC file.
+   */
+  private final Map<String, String> ndcProductHashMap = new HashMap<>();
+
+  /** Tracks the national drug codes that have already had code lookup failures. */
+  private final Set<String> drugCodeLookupMissingFailures = new HashSet<>();
+
+  /**
+   * Cached copy of the testing version of the {@link FdaDrugCodeDisplayLookup} so that we don't
+   * have to construct it over and over in the unit tests
+   */
+  private static FdaDrugCodeDisplayLookup drugCodeLookupForTesting;
+
+  /**
+   * Cached copy of the production version of the {@link FdaDrugCodeDisplayLookup} so that we don't
+   * have to construct it over and over in the unit tests
+   */
+  private static FdaDrugCodeDisplayLookup drugCodeLookupForProduction;
+
+  /**
+   * Factory method for creating a {@link FdaDrugCodeDisplayLookup} for testing that includes the
+   * fake drug code.
+   *
+   * @return the {@link FdaDrugCodeDisplayLookup}
+   */
+  public static FdaDrugCodeDisplayLookup createDrugCodeLookupForTesting() {
+    if (drugCodeLookupForTesting == null) {
+      drugCodeLookupForTesting = new FdaDrugCodeDisplayLookup(true);
+    }
+
+    return drugCodeLookupForTesting;
+  }
+
+  /**
+   * Factory method for creating a {@link FdaDrugCodeDisplayLookup} for production that does not
+   * include the fake drug code.
+   *
+   * @return the {@link FdaDrugCodeDisplayLookup}
+   */
+  public static FdaDrugCodeDisplayLookup createDrugCodeLookupForProduction() {
+    if (drugCodeLookupForProduction == null) {
+      drugCodeLookupForProduction = new FdaDrugCodeDisplayLookup(false);
+    }
+
+    return drugCodeLookupForProduction;
+  }
+
+  /**
+   * Constructs an {@link FdaDrugCodeDisplayLookup}
+   *
+   * @param includeFakeDrugCode whether to include the fake testing drug code or not
+   */
+  private FdaDrugCodeDisplayLookup(boolean includeFakeDrugCode) {
+    readFDADrugCodeFile(includeFakeDrugCode);
+  }
```

Comment:
Nice to move this to eager loading, rather than lazy loading (which I _think_ it was before?). Thanks.